pkg/didcomm/crypto/legacy/authcrypt/authcrypt.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package authcrypt
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/agl/ed25519/extra25519"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/nacl/box"
+)
+
+type privateEd25519 [ed25519.PrivateKeySize]byte
+type publicEd25519 [ed25519.PublicKeySize]byte
+
+type keyPairEd25519 struct {
+	priv *privateEd25519
+	pub  *publicEd25519
+}
+
+// CurveKeySize is the size of public and private Curve25519 keys in bytes
+const CurveKeySize int = 32
+
+type privateCurve25519 [CurveKeySize]byte
+type publicCurve25519 [CurveKeySize]byte
+
+type keyPairCurve25519 struct {
+	priv *privateCurve25519
+	pub  *publicCurve25519
+}
+
+// Crypter represents an Authcrypt Encrypter (Decrypter) that outputs/reads legacy Aries envelopes
+type Crypter struct {
+	sender     keyPairEd25519
+	recipients []*publicEd25519
+	randSource io.Reader
+}
+
+// New will create a Crypter that encrypts messages using the legacy Aries format
+// Note: legacy crypter does not support XChacha20Poly1035 (XC20P), only Chacha20Poly1035 (C20P)
+func New(sender keyPairEd25519, recipients []*publicEd25519) (*Crypter, error) { // nolint: lll
+
+	if len(recipients) == 0 {
+		return nil, errors.New("empty recipients keys, must have at least one recipient")
+	}
+
+	c := &Crypter{
+		sender:     sender,
+		recipients: recipients,
+		randSource: rand.Reader,
+	}
+
+	if !isKeyPairValid(sender) {
+		return nil, fmt.Errorf(
+			"sender keyPair not supported, it must have a %d byte private key and %d byte public key",
+			ed25519.PrivateKeySize, ed25519.PublicKeySize)
+	}
+
+	return c, nil
+}
+
+func (c *Crypter) setRandSource(source io.Reader) {
```

Comment:
This method is unused

Reply:
It's used in tests, but I can replace it with direct access to the Crypter.